packages/firefuel/lib/src/rules.dart

```diff
@@ -21,18 +21,29 @@ abstract class CollectionRead<R, T extends Serializable> {
   /// See: StreamBuilder
   Stream<R> listenAll();
 
+  /// Get up to the maximum number of documents specified by the [limit]
+  ///
+  /// Returns 0 to [limit]
+  Future<R> limit(int limit);
+
+  /// Get up to the maximum number of documents specified by the [limit]
+  ///
+  /// Refreshes automatically when new data is added/removed from the
+  /// collection
+  Stream<R> listenLimited(int limit);
+
   /// Get a list of documents matching all clauses
   ///
   /// throws a [MissingValueException] when no [Clause]s are given
-  Future<R> where(List<Clause> clauses);
+  Future<R> where(List<Clause> clauses, {int? limit});
 
   /// Get a list of documents matching all clauses
   ///
   /// Refreshes automatically when new matching data is added/removed from the
   /// collection
   ///
   /// throws a [MissingValueException] when no [Clause]s are given
-  Stream<R> listenWhere(List<Clause> clauses);
+  Stream<R> listenWhere(List<Clause> clauses, {int? limit});
 }
```

Comment:
We need to add comments here about `[limit]`

Reply:
Might as well. The more documentation there is the less questions we'll get (hopefully 🤞 )